kotlinx-coroutines-core/common/src/flow/internal/NullSurrogate.kt

```diff
@@ -4,11 +4,11 @@
 
 package kotlinx.coroutines.flow.internal
 
+import kotlinx.coroutines.internal.*
 import kotlin.jvm.*
 
-internal object NullSurrogate {
-
-    @JvmStatic
-    @Suppress("UNCHECKED_CAST")
-    internal fun <T> unbox(value: Any?): T = if (value === NullSurrogate) null as T else value as T
-}
+// Note: it is conceptually the same as AbstractChannel.NULL_VALUE
+// todo: consolidate, move this constant to "common" kotlinx.coroutines.internal, rename consistently with other symbols
```

Comment:
About todo, there is always a chance to accidentally mix two unrelated symbols (as we did with `Incompleted`), so I'd prefer it so stay into its own package. Renaming it to `NULL_VALUE` is reasonable